src/data.jl

```diff
@@ -540,8 +544,10 @@ function smear(
     dos::DensityOfStates,
     sigma::Real
 )
-    smear = ifft(fft(dos.dos) .* gaussian.(sigma, energies(dos)))
-    return DensityOfStates(fermi(dos), dos.dos, abs.(smear))
+    # Convolute DOS with gaussian using Fourier transforms
+    gsmear = real(ifft(fft(dos.dos) .* fft(gaussian(dos, sigma))))
+    # Normalize gaussian smearing so that sum(gaussian smear) = sum(DOS)
+    return DensityOfStates(fermi(dos), energies(dos), (gsmear)*(sum(dos.dos)/sum(gsmear)))
```

Comment:
Have you tried seeing whether the smearing significantly affects the number of electrons? I believe `nelectrons(::DensityOfStates)` exists that integrates the DOS up to the fermi energy.